src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx

```diff
@@ -646,31 +329,16 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
         throw new Error(`THORChain pool halted for assetId: ${assetId}`)
       }
 
-      const maybeTxId = await (async () => {
-        if (isTokenWithdraw) {
-          return handleCustomTx()
-        }
-
-        const withdrawInput = await getWithdrawInput()
-        if (!withdrawInput) throw new Error('Error building send input')
-        return handleMultiTxSend()
-      })()
-
-      if (!maybeTxId) {
-        throw new Error('Error sending THORCHain savers Txs')
-      }
+      const _txId = await executeTransaction()
+      if (!_txId) throw new Error('No txId returned from onSignTx')
```

Comment:
This seems to be repeated several times, should we just make `executeTransaction` throw when unable to produce a txId?

Reply:
I'm not sure we want to throw from `executeTransaction` itself - checking for `txId` presence (or lack thereof) is pretty standard flow for us - also consumers may not handle failures explictly, vs. an explicit `undefined` txId which *has* to be handled from consumers (to e.g produce side-effects like setting isSubmitting to `false)